content/general/concepts/what-is-the-singleton-pattern/what-is-the-singleton-pattern.md

```diff
@@ -0,0 +1,45 @@
+---
+Title: 'What is the Singleton Pattern?'
+Description: 'The singleton pattern ensures a class has one, and only one object instantiated during the programs lifecycle. It is used to guarantee the control of a resource through its implementation.'
+Subjects:
+  - 'Computer Science'
+  - 'Interview Prep'
+Tags:
+  - 'Classes'
+  - 'Conceptual'
+  - 'Constructors'
+  - 'Objects'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The singleton pattern ensures a class has one, and only one object instantiated during the program's lifecycle. It is used to guarantee the control of a resource through its implementation.
+
+## UML Design
+
+![UML diagram of a singleton](media/singleton-uml.png)
+
+## Implementation
+
```

Comment:
In terms of pedagogy, it may be beneficial to explain the code before the actual code snippet.

While comments within code snippets are convenient, let's try moving some larger chunks of comments above the code.